src/main/java/net/rptools/maptool/client/ui/MapToolFrame.java

```diff
@@ -1519,6 +1545,7 @@ private void stopTokenDrag() {
   public void setCurrentZoneRenderer(ZoneRenderer renderer) {
     // Flush first so that the new zone renderer can inject the newly needed images
     if (renderer != null) {
+      updatePlayerStatus(renderer.getZone());
```

Comment:
There is a ZoneDeactivated put onto the event bus when a zone becomes deactivated, 
See HTMLFrameFactory.java for how to subscribe and react to this event. This is the preferred method for reacting to zone changes. You might need to add a new event ZoneLoading for this though. This will also allow others to subscribe to the event in future rather than having to find the right line to add the code.

Reply:
Where should I put the listeners to these new events that would make changes to the LocalPlayer?

Reply:
> Where should I put the listeners to these new events that would make changes to the LocalPlayer?

Create a new class in the player package called `PlayerZoneListener` (or something more appropriate if it listens to other events)

Then in `MapTool.initialize()`  register the listener between the create of the player (line 670) and the creation of the basic campaign (line 671).
https://github.com/RPTools/maptool/blob/2586c63de633919c275945a6d87d556c598b2ab8/src/main/java/net/rptools/maptool/client/MapTool.java#L669-L674

Reply:
Looks like ZoneDeactivated isn't always sent i.e. I saw two ZoneActivated in a row so decide to use the new zoneLoading zoneLoaded events